source/common/http/conn_manager_impl.cc

```diff
@@ -1640,9 +1640,18 @@ void ConnectionManagerImpl::ActiveStream::encodeHeaders(ActiveStreamEncoderFilte
 void ConnectionManagerImpl::ActiveStream::encodeHeadersInternal(ResponseHeaderMap& headers,
                                                                 bool end_stream) {
   // Base headers.
-  if (!stream_info_.hasResponseFlag(StreamInfo::ResponseFlag::ResponseFromCacheFilter)) {
+
```

Comment:
Please also add a release note of the change, including the runtime guard, to docs/root/version_history/current.rst